ichnaea/data/tests/test_export.py

```diff
@@ -452,31 +472,32 @@ def test_stats(self, celery, session, stats):
         self.add_reports(celery, 3, api_key=None)
         self._update_all(session)
 
-        stats.check(
-            counter=[
-                ("data.export.batch", 1, 1, ["key:internal"]),
-                ("data.report.upload", 2, 3),
-                ("data.report.upload", 1, 3, ["key:test"]),
-                ("data.report.upload", 1, 6, ["key:e5444-794"]),
-                ("data.observation.upload", 1, 3, ["type:cell", "key:test"]),
-                ("data.observation.upload", 1, 6, ["type:wifi", "key:test"]),
-                ("data.observation.upload", 0, ["type:cell", "key:no_key"]),
-                ("data.observation.upload", 1, 6, ["type:cell", "key:e5444-794"]),
-                ("data.observation.upload", 1, 12, ["type:wifi", "key:e5444-794"]),
-            ]
-        )
-        # we get a variable number of statsd messages and are only
-        # interested in the sum-total
-        for name, total in (("cell", 12), ("wifi", 24)):
-            insert_msgs = [
-                msg
-                for msg in stats.msgs
-                if (msg.startswith("data.observation.insert") and "type:" + name in msg)
-            ]
-            assert (
-                sum([int(msg.split(":")[1].split("|")[0]) for msg in insert_msgs])
-                == total
-            )
```

Comment:
The way I rewrote this covers both sets of assertions. It asserts the batch and upload messages are covered and by which keys. It also asserts the total number of inserted items by type.